services/QuillLMS/client/app/bundles/Staff/components/comprehension/changeLog/changeLog.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from "react";
+import * as _ from 'underscore';
+import { RouteComponentProps } from 'react-router-dom'
+import { useQuery } from 'react-query';
+import { firstBy } from "thenby";
+import ReactTable from 'react-table';
+import qs from 'qs';
+import * as _ from 'lodash'
+import DateTimePicker from 'react-datetime-picker';
+
+import { renderHeader } from '../../../helpers/comprehension';
+import { sort } from '../../../../../modules/sortingMethods.js';
+import { fetchChangeLog, fetchActivity } from '../../../utils/comprehension/activityAPIs';
+import { DropdownInput, Spinner, } from '../../../../Shared/index';
+
+interface ChangeLogProps {
+
+}
+
+const ChangeLog = ({ history, match }) => {
+  const { params } = match;
+  const { activityId, } = params;
+  const SESSION_INDEX=1;
+  const initialStartDateString = window.sessionStorage.getItem(`${SESSION_INDEX}startDate`) || '';
+  const initialEndDateString = window.sessionStorage.getItem(`${SESSION_INDEX}endDate`) || '';
+  const initialStartDate = initialStartDateString ? new Date(initialStartDateString) : null;
+  const initialEndDate = initialEndDateString ? new Date(initialEndDateString) : null;
+
+  const [searchInput, setSearchInput] = React.useState<string>('');
+  const [prompt, setPrompt] = React.useState<string>('all');
+  const [rule, setRule] = React.useState<string>('all');
+  const [startDate, onStartDateChange] = React.useState<Date>(initialStartDate);
+  const [endDate, onEndDateChange] = React.useState<Date>(initialEndDate);
+
+  const { data: changeLogData, status: status } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
```

Comment:
I think we probably want a different query key here that is more specific to this data?